File: 950. Reveal Cards In Increasing Order/Solution.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);

        int n = deck.length; 
        int[] ans = new int[n];

        Queue<Integer> queue = new LinkedList<>();

        for(int i=0;i<n;i++){
            queue.offer(i);
        }
        int index = 0;
        while(!queue.isEmpty() && index<n){
            ans[queue.poll()] = deck[index++];
            queue.offer(queue.poll());
        }
        
        return ans;
    }
}

File: 950. Reveal Cards In Increasing Order/Solution.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        n = len(deck)
        res = [0] * n

        queue = deque(range(n))
        deck.sort()
        for n in deck:
            res[queue.popleft()] = n
            if queue:
                queue.append(queue.popleft())
        
        return res